Sources/MacroToolkit/Modifiers/Modifier.swift

```diff
@@ -0,0 +1,23 @@
+import SwiftSyntax
+
+/// A protocol for modifiers in Swift that are represented by `TokenKind`.
+public protocol Modifier: RawRepresentable where RawValue == TokenKind {
+    /// Initializes a `Modifier` from a list of declaration modifiers.
+    ///
+    /// - Parameter modifiers: A list of declaration modifiers.
+    init?(modifiers: DeclModifierListSyntax)
+}
+
+extension Modifier {
+    /// Default implementation for initializing a `Modifier` from a list of declaration modifiers.
+    ///
+    /// - Parameter modifiers: A list of declaration modifiers.
+    public init?(modifiers: DeclModifierListSyntax) {
```

Comment:
Perhaps provide a `firstModifierOfKindIn` parameter label to make the behaviour a bit more clear. Otherwise it's a bit strange initialising a single modifier from a list of modifiers.